pkg/fleet-manager/fleet_plugin.go

```diff
@@ -47,30 +48,33 @@ func (f *FleetManager) reconcilePlugins(ctx context.Context, fleet *fleetapi.Fle
 			err        error
 		}
 
-		resultsChannel := make(chan reconcileResult, 4)
+		type reconcileFunc func(context.Context, *fleetapi.Fleet, map[ClusterKey]*fleetCluster) (kube.ResourceList, ctrl.Result, error)
 
-		go func() {
-			result, ctrlResult, err := f.reconcileMetricPlugin(ctx, fleet, fleetClusters)
-			resultsChannel <- reconcileResult{result, ctrlResult, err}
-		}()
+		funcs := []reconcileFunc{
+			f.reconcileMetricPlugin,
+			f.reconcileGrafanaPlugin,
+			f.reconcileKyvernoPlugin,
+			f.reconcileBackupPlugin,
+		}
 
-		go func() {
-			result, ctrlResult, err := f.reconcileGrafanaPlugin(ctx, fleet)
-			resultsChannel <- reconcileResult{result, ctrlResult, err}
-		}()
+		resultsChannel := make(chan reconcileResult, len(funcs))
+		var wg sync.WaitGroup
 
-		go func() {
-			result, ctrlResult, err := f.reconcileKyvernoPlugin(ctx, fleet, fleetClusters)
-			resultsChannel <- reconcileResult{result, ctrlResult, err}
-		}()
+		for _, fn := range funcs {
+			wg.Add(1)
+			go func(fn reconcileFunc) {
+				defer wg.Done()
+				result, ctrlResult, err := fn(ctx, fleet, fleetClusters)
+				resultsChannel <- reconcileResult{result, ctrlResult, err}
+			}(fn)
+		}
 
 		go func() {
-			result, ctrlResult, err := f.reconcileBackupPlugin(ctx, fleet, fleetClusters)
-			resultsChannel <- reconcileResult{result, ctrlResult, err}
+			wg.Wait()
+			close(resultsChannel)
 		}()
 
-		for i := 0; i < 4; i++ {
-			res := <-resultsChannel
+		for res := range resultsChannel {
 			if res.err != nil || res.ctrlResult.RequeueAfter > 0 {
```

Comment:
We should aggregate the error result here

Reply:
ok